src/components/server/ServerManager.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  CommandBar,
+  ICommandBarItemProps,
+  INavLink,
+  Nav,
+  PrimaryButton,
+  Stack,
+} from "@fluentui/react";
+import React, { FC, useEffect, useState } from "react";
+import Server, { SERVER_PREFIX } from "../../types/server";
+import { getItems, saveItem, deleteItem } from "../../storage/storage";
+import ServerEdit from "./ServerEdit";
+import ServerDeleteConfirmation from "./ServerDeleteConfirmation";
+import ServerConnectionError from "./ServerConnectionError";
+import { stackTokens, serverPanel } from "../../style";
+
+type ServerManagerProps = {
+  onConnect: (server: Server) => Promise<void>;
+};
+
+function replaceAtIndex<T>(array: T[], item: T, index: number) {
+  return Object.assign([], array, { [index]: item });
+}
+
+function removeAtIndex<T>(array: T[], index: number) {
+  return array.filter((_, i) => i !== index);
+}
+
+const ServerManager: FC<ServerManagerProps> = ({ onConnect }) => {
+  const [servers, setServers] = useState<Server[]>([]);
+  const [selectedServerId, setSelectedServerId] = useState("");
+  const [connectionError, setConnectionError] = useState("");
+  const [hideDeleteConfirmation, setHideDeleteConfirmation] = useState(true);
+
+  async function loadServers() {
+    const items = (await getItems(SERVER_PREFIX)) as Server[];
+    setServers(items);
+  }
+
+  useEffect(() => {
+    loadServers();
+  }, []);
+
+  const groups = [
```

Comment:
I think we'll need to wrap this in an useEffect to update it otherwise I don't think it will update when we add or remove a server

Reply:
No, this works just fine - each time the component re-renders, the `ServerManager` function is invoked causing this code to be executed.